frankAllSkyCam/config.txt
########################################
# frankAllSkyCam
# configuration file -  config.txt
#
# please adjust according to your needs
# see comments and documentation
########################################

[system]
# where will be stored images - relative path to frankAllSkyCam
otuputFolder= img

# directory where to save logs
logFolder = log

# local web folder where to save the skycam.jpg
outputLocalWebFile = /var/www/html/img/skycam.jpg

#how many days you want to keep jpgs
days_retention = 3

#watchdog will reboot if after x minutes no image has been generated
rebootAfter = 15

[site]
# configure your site (lat / long / elevation in meters, pytz timezone)
latitude = 44.75
longitude = 9.29
elevation = 1150
time_zone = Europe/Rome
#the following label will be printed on the image
inte = <your_name_here>


[resolution]
# may use  800 x 600 as you wish
horiz = 1024
vert = 768
picture_rotation = 0

[libcamera]
# may personalize the libcamara params.
# please DO NOT set --shutter, --immediate
# night parameters will be used only when exposure > 0
additional_night_params = --gain 14 --awbgains 2.7,1.5 --contrast 1.5
additional_day_params = --metering average


[timelapse]
# if True, timelapse from sunset to sunrise will be generated  Otherwise, not.
nightTL = True

# if True, 24h timelapse from 9am to 9am next day will be generated. Otherwise, not.
fullTL = True
nightOnly = True

# following parameters are used by ffmpeg
tl_horiz = 1024
tl_vert = 768
framerate = 15
ffmpeg1 = -c:v libx264 -crf 12 -preset slow -pix_fmt yuv420p
# additional ffmpeg parameters may be added by experts!
ffmpeg2 =
ffmpeg3 =

[font]
#when 1024x768 you can increase it
font_size = 20
font_colorR = 0
font_colorG = 255
font_colorB = 255

[text_coords]
# coordinates of the text to be added on the allskycam image
  #date,time
top_left_x = 20
top_left_y = 0

 # moon
top_right_x = 820
top_right_y = 0

 #night and SQM
bottom_left_x = 20
bottom_left_y = 715

 #sun
bottom_right_x = 870 
bottom_right_y = 710

  #top, center (your name, see the [site] section above)
top_center_x = 450
top_center_y = 0


[extra_text]
# et_use = y if you want to add some extra text
et_use= y
# absolute path to your txt file
et_data_file = /home/pi/frankAllSkyCam/extra_data.txt
et_font_size = 20
et_font_colorR = 0
et_font_colorG = 255
et_font_colorB = 255
et_x_pos = 20
et_y_pos = 90

[compass]
# relative path from frankAllSkyCam/png
# if file not found, it will be skipped
# set your coords according to your resolution
compass_filename = compass.png
compass_x_pos = 915
compass_y_pos = 610
compass_rot_angle = 0

[logo]
# relative path from frankAllSkyCam/png
# if file not found, it will be skipped
# set your coords according to your resolution
logo_filename = logo.png
logo_x_pos = 30
logo_y_pos = 610

[moon_phase_img]
# relative path from frankAllSkyCam/png
# if file not found, it will be skipped
# set your coords according to your resolution
moon_filename = phase.png
moon_x_pos = 920
moon_y_pos = 110

[planets]
# relative path from frankAllSkyCam/png
# if file not found, it will be skipped
# set your coords according to your resolution

jupiter_x_pos = 15
mars_x_pos = 55
saturn_x_pos = 90
venus_x_pos = 142

jupiter_y_pos = 52
mars_y_pos = 53
saturn_y_pos = 52
venus_y_pos = 52


[exposure]
# max night exposure in seconds
#  for urban sky, suggested value is 4
#  for dark sky, suggested value is 55-60 (I found 55 secs ok for sqm 21,3).
#  this value only limits the max exposure. The calculated or real SQM will drive exposure
esp_secs =60

[ftp]
# if True, allSkyCam image, timelapse and startrails
#will be uploaded on your FTP Server
isFTP=False

# ftp parameters, to upload the allSkyCam image to your FTP server
FTP_server=your_server_here.com
FTP_login=your_username
FTP_pass= your_password
FTP_uploadFolder =/your_folder/

# uploaded file name (e.g.,allskycam.jpg). no need to add the jpg extention
FTP_filenameAllSkyImgJPG = /img/allskycam

# allskycam_night.mp4 will be generated if nightTL = True
# allskycam_24h.mp4 will be generated if fullTL = True
FTP_fileNameTimelapseMP4 = /video/frankAllSkycam

# starTrail.jpg will be generated
FTP_fileNameStarTrailJPG = /startrail/startrail.jpg

[sqm]
# info about your SQM-LE reader (if you have not it, just let use_sqm_le = n)
# directory where to save sqm info
sqmFolder = sqm
# would you use the SQM-LE ? y/n
use_sqm_le = n
# would you log your SQM ? y/n
sqmLog = n
# keep sqmDebug = n
sqmDebug = n
# in case you have SQM-LE, put your IP/port
ip_address = 192.168.2.59
port = 10001



frankAllSkyCam/helper/config.txt
########################################
# frankAllSkyCam
# configuration file -  config.txt
# 
# please adjust according to your needs
# see comments and documentation
########################################

[system]
# where will be stored images from all Sky cam ?
otuputFolder= /home/pi/frankAllSkyCam/img
# 
# directory where to save logs
logFolder = /home/pi/frankAllSkyCam/log

# directory  where to save SQM LE readings
sqmFolder = /home/pi/frankAllSkyCam/sqm

# web folder where to save the skycam.jpg
outputLocalWebFile = /var/www/html/img/skycam.jpg

#how many days you want to keep jpgs
days_retention = 3

#watchdog will reboot if after 15 minutes no /home/pi/frankAllSkyCam/log/alive.txt has been generated 
rebootAfter = 15

[site]
#the following label will be printed on the image
inte = www.astrobrallo.com
latitude = 44.73
longitude = 9.31
time_zone = Europe/Rome

[resolution]
# may use  1024 x 768
horiz = 800
vert = 600
picture_rotation = 0

[libcamera]
# may personalize the libcamara params. 
# please DO NOT set --shutter, --gain, --awbgains --immediate
additional_params = ""

[timelapse]
# if True, timelapse from sunset to sunrise will be generated  Otherwise, not.
nightTL = True

# if True, 24h timelapse from 9am to 9am next day will be generated. Otherwise, not. 
fullTL = True
nightOnly = True

# following parameters are used by ffmpeg
tl_horiz = 800
tl_vert = 600
framerate = 10
ffmpeg1 = -c:v libx264 -crf 12 -preset slow -pix_fmt yuv420p 
# additional ffmpeg parameters may be added by experts!
ffmpeg2 = 
ffmpeg3 = 
 
[font]
#when 1024x768 you can increase it
font_size = 18

[exposure]
#night exposure in seconds
# urban sky:
#esp_secs = 4
# dark sky (i found 55 secs ok for sqm 21,3).
# adjust according to your sky
esp_secs =60
# max exposure during dawn and dusk:
esp_dawn_dusk = 40 

[moon]
#exposure reduction because of the moon
full_moon_reduc = 0.75

# minutes after moonset
ms_offset = -30

# minutes from moonset-start to no-moon
ms_duration = 50      

# minutes *before* the moon rises
mr_offset = -30 

# minutes from moon-rise to moon fully up
mr_duration = 60


[offset]
# minutes to offset dawn and dusk for optimal exposure 
dawn = 0
dusk = 0

[ftp]
# if True, allSkyCam image, timelapse and startrails 
#will be uploaded on your FTP Server 
isFTP=True

# ftp parameters allSkyCam image will be uploaded on a 
FTP_server = myftpserver.com
FTP_login = myuser
FTP_pass = mypass
FTP_uploadFolder =/public/frankAllSkyCam/

# allskycam.jpg will be generated on the FTP server
FTP_filenameAllSkyImgJPG = allskycam 

# allskycam_night.mp4 will be generated if nightTL = True
# allskycam_24h.mp4 will be generated if fullTL = True
FTP_fileNameTimelapseMP4 = /videos/frankAllSkycam

# starTrail.jpg will be generated 
FTP_fileNameStarTrailJPG = /startrails/starTrail.jpg

[sqm_le]
# if you want to use SQM LE then use_sqm = y Otherwise, not.
use_sqm = n
ip_address = 192.168.2.59
port = 10001
write_log = y



frankAllSkyCam/config.py
########################################
# frankAllSkyCam
# configuration file -  config.txt
#
# please adjust according to your needs
# see comments and documentation
########################################

[system]
# where will be stored images - relative path to frankAllSkyCam
otuputFolder= img

# directory where to save logs
logFolder = log

# local web folder where to save the skycam.jpg
outputLocalWebFile = /var/www/html/img/skycam.jpg

#how many days you want to keep jpgs
days_retention = 3

#watchdog will reboot if after x minutes no image has been generated
rebootAfter = 15

[site]
#the following label will be printed on the image
inte = www.astrobrallo.com
latitude = 44.73
longitude = 9.31
time_zone = Europe/Rome

[resolution]
# may use  1024 x 768
horiz = 800
vert = 600
picture_rotation = 0

[libcamera]
# may personalize the libcamara params.
# please DO NOT set --shutter, --immediate
# night parameters will be used only when exposure > 0
additional_night_params = --gain 14 --awbgains 2.2,1.6 --contrast 1.5
additional_day_params = --metering average


[timelapse]
# if True, timelapse from sunset to sunrise will be generated  Otherwise, not.
nightTL = True

# if True, 24h timelapse from 9am to 9am next day will be generated. Otherwise, not.
fullTL = True
nightOnly = True

# following parameters are used by ffmpeg
tl_horiz = 800
tl_vert = 600
framerate = 10
ffmpeg1 = -c:v libx264 -crf 12 -preset slow -pix_fmt yuv420p
# additional ffmpeg parameters may be added by experts!
ffmpeg2 =
ffmpeg3 =

[font]
#when 1024x768 you can increase it
font_size = 18
font_colorR = 0
font_colorG = 255
font_colorB = 255

[text_coords]
# coordinates of the text
  #date,time
top_left_x = 5
top_left_y = 0

 # sunrise / sunset
top_right_x = 670
top_right_y = 0

 #night and SQM
bottom_left_x = 5
bottom_left_y = 535
 #moon
bottom_right_x = 630
bottom_right_y =  535

  #top, center (your name)
top_center_x = 300
top_center_y = 0


[extra_text]
# et_use = y if you want to add
et_use= n
# absolute path to your txt file
et_data_file = /home/pi/frankAllSkyCam/extra_text.txt
et_font_size = 18
et_font_colorR = 0
et_font_colorG = 255
et_font_colorB = 255
et_x_pos = 5
et_y_pos = 80

[compass]
# relative path from frankAllSkyCam
# if file not found, it will be skipped
# set your coords according to your resolution
compass_filename = compass.png
compass_x_pos = 900
compass_y_pos = 580
compass_rot_angle = 20

[logo]
# relative path from frankAllSkyCam
# if file not found, it will be skipped
# set your coords according to your resolution
logo_filename = logo.png
logo_x_pos = 30
logo_y_pos = 580

[exposure]
# max night exposure in seconds
#  for urban sky, suggested value is 4
#  for dark sky, suggested value is 55-60 (I found 55 secs ok for sqm 21,3).
#  this value only limits the max exposure. The calculated or real SQM will drive exposure
esp_secs =60

[offset]
# minutes to offset dawn and dusk for optimal exposure
dawn = 0
dusk = 0

[ftp]
# if True, allSkyCam image, timelapse and startrails
#will be uploaded on your FTP Server
isFTP=False

# ftp parameters, to upload the allSkyCam image to your FTP server
FTP_server=yourftpserver.com
FTP_login=username
FTP_pass=password
FTP_uploadFolder =/public_html

# uploaded file name (e.g.,allskycam.jpg)
FTP_filenameAllSkyImgJPG = img/allskycam

# allskycam_night.mp4 will be generated if nightTL = True
# allskycam_24h.mp4 will be generated if fullTL = True
FTP_fileNameTimelapseMP4 = /video/frankAllSkycam

# starTrail.jpg will be generated
FTP_fileNameStarTrailJPG = /startrail/startrail.jpg

[sqm]
# directory where to save sqm info
sqmFolder = sqm
# would you use the SQM-LE ? y/n
use_sqm_le = n
# would you log your SQM ? y/n
sqmLog = n
# keep sqmDebug = n
sqmDebug = n
# in case you have SQM-LE, put your IP/port
ip_address = 192.168.2.59
port = 10001

